pywt/tests/test__pywt.py

```diff
@@ -108,6 +108,17 @@ def test_dwt_max_level():
     assert_(pywt.dwt_max_level(16, 10) == 0)
     assert_(pywt.dwt_max_level(16, 18) == 0)
 
+    # accepts discrete Wavelet object or string as well
+    assert_(pywt.dwt_max_level(32, pywt.Wavelet('sym5')) == 1)
+    assert_(pywt.dwt_max_level(32, 'sym5') == 1)
+
+    # string input that is not a discrete wavelet
+    assert_raises(TypeError, pywt.dwt_max_level, 16, 'mexh')
```

Comment:
This should be ``ValueError`` with an understandable error message (`` `filter_len` is not the name of a discrete wavelet`` for example.) 

Maybe put ``elif filter_len in wavelist(kind='discrete'):`` inside an ``isinstance(filter_len, string_types):``? I see we don't have ``string_types`` yet; it can be stolen from ``six`` if needed.